### DISCUSSION TITLE
Why is the .java extension mandatory when using `jbang init` ?

### DISCUSSION THREAD

**Discussion by jmini** [jbangdev/jbang#676] [Q&A] [answered]
Last time I used `jbang`, I just did something like:

```
jbang init fooBar
```

This works great and the file can be executed with:

```
jbang fooBar
```

(as indicated by the tool: `[jbang] File initialized. You can now run it with 'jbang fooBar'`. This is great.)

Not having the `.java` extension is not ideal when you look at the file later.

---

I was wondering why the `.java` extension is not added to the file.

E.g create a file `fooBar.java` instead of `fooBar` and when `jbang fooBar` is executed: first look for the presence of `fooBar` and if not found, retry with `fooBar.java` before failing.

If there is a real use case for this file without any extension, at least it should be added when `edit` is executed.

**Comment by maxandersen** [comment]
I assume you meant to ask "Why is the .java extension *not* mandatory?" ?

it is not mandatory so you can use the script as if it was a shell script - some people get really caught up by files being named .py, .sh, .java, etc. when they can just be named what they do and then have the used language being a implementation detail.


> If there is a real use case for this file without any extension, at least it should be added when `edit` is executed.

when you do `jbang edit test` it will symbolically link it up as `test.java` to have it work in the editor.

I'm not sure what you are seeing as an issue as it seems to me it works exactly how you ask ?

> E.g create a file `fooBar.java` instead of `fooBar` and when `jbang fooBar` is executed: first look for the presence of `fooBar` and if not found, retry with `fooBar.java` before failing.

I think with jbang alias and catalogs we have quite a powerful mechanism to provide shorthand names.

Adding one more here, especially for a *physical* named thing you already explicitly created using the short name "fooBar" feels "too much" imo. i.e. I don't see a compelling use case for it.

**Comment by jmini** [comment]
Thank you for the quick reply. Sorry if my message was not clear enough.

Let me simplify what I mean:

When I run:
```
jbang init fooBar
```

I expect the file `fooBar.java` to be created.
And the message to be `[jbang] File initialized. You can now run it with 'jbang fooBar.java'`

**Comment by maxandersen** [comment, marked as answer]
Gotcha - the current behaviour is intentional. See https://github.com/jbangdev/jbang#extension-lessnon-java-files-for-cli-plugins

in short - if jbang forced `jbang init kubectl-foobar` to be `kubectl-foobar.java` you can't use jbang scripts to allow `kubectl foobar` since it will be loooking for `kubectl-foobar` file in the `PATH` not `kubectl-foobar.java`
